### PR TITLE
feat(vanilla-extract): Converting a component to vanilla-extract

### PR DESCRIPTION
**Pull request by Benrajalu**
**What is the problem this PR is trying to solve?**

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Changelog has been commited, e.g: `yarn changeset`
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by frassinier** [comment]
Linked to #268

File: src/components/Accordion/Accordion.css.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+});
+
+export const item = style({
+	margin: `${tokens.space.m} 0`,
+	minWidth: '25rem',
+	border: `1px solid ${theme.colors.accordionBorderColor}`,
+	borderRadius: tokens.radii.rectRadius,
+
+	':hover': {
+		borderColor: tokens.colors.gray[200],
```
Comment (frassinier) [inline src/components/Accordion/Accordion.css.ts:17]: this must come from the theme, from an alias

File: src/components/Button/Button.css.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+});
+
+export const buttonWithText = style({
+	[`${smallButton} &`]: {
+		padding: 0,
+	},
+});
+
+const baseHoverStyles = {
+	color: `var(--t-button-color, ${theme.colors?.textColor})`,
```
Comment (frassinier) [inline src/components/Button/Button.css.ts:38]: why keeping CSS vars here?

File: src/components/Accordion/Accordion.css.ts (hunk 1 above)
Comment (Benrajalu) [reply src/components/Accordion/Accordion.css.ts:17]: Agreed, I'm only copy-pasting the current styles right now

**Pull request closed without merging**